internal/common/validator.go

```diff
@@ -67,9 +66,38 @@ func ValidateRequestEntity(ctx *fiber.Ctx, request interface{}, errorMessage str
 
 	if err := ValidateStruct(request); err != nil {
 		return ErrorWithValidationErrors(
-			fiber.StatusUnprocessableEntity, errorMessage, "invalid format", err,
+			fiber.StatusUnprocessableEntity, errorMessage, err,
 		)
 	}
 
 	return nil
 }
+
+func GenerateErrorDetails(validationErrors []ValidationError) string {
+	var errors []string
+
+	for _, validationError := range validationErrors {
+		switch validationError.Field {
```

Comment:
I'd use `Tag()` here instead of hardcoding the fields.

That way we can keep the validation tags on the structs as the source of truth and won't need to duplicate the information here. Also that would allow for differentiating multiple validation errors on the same field (fe. `description` being having size limits AND allowed characters limits).

There's an [example here](https://github.com/go-playground/validator/issues/559) - (validator [also has translations](https://github.com/go-playground/validator/blob/master/_examples/translations/main.go) for changing error messages, but I think they're too much for what we are doing)

Reply:
>  [...] instead of hardcoding the fields.

Yeah, not a fan of hardcoding too, though I had to use it because am not familiar with the Validation Library you're using so I didn't know about the `Tag()` function.

I'm working on a new solution based on the example you provided me right now. 🙂